### DISCUSSION TITLE
-1 label after preprocessing

### DISCUSSION THREAD

**Discussion by Juusotak** [MIC-DKFZ/nnUNet#2499] [Q&A]
I inspected my dataset after running nnUNets preprocessing steps. At some segmentation files there is added label with value -1. My dataset only has labels 0 for background and label 1 for the lesion of interest. Is this -1 label expected behaviour or is this some kind of error during the preprocessing?

**Comment by fhrzic** [comment]
Hello,

It is expected behavior. Namely when sampling random location of the patch, the -1 is given to all places in segmentation map that fall outside the patch. Check script dataloader_3d.py in nnuntv2/training/dataloading. 

I hope this helped.
Franko
